### PR TITLE
AngularMigration: Add migration button to old graph panel 

### PR DESCRIPTION
**Pull request by torkelo**
* [x] Adds migration button to old graph (the migration logic was already there).
* [x] Set state to "deprecated" to hide it from being selected in the visualization selection

### CODE REVIEW

**Comment by leeoniya** [comment]
nervous about this.

a few remaining things like missing transform negative, tooltip and legend sort order are widely regarded as regressions rather than feature requests, with many stating they cannot use Time series as a replacement without them.

https://github.com/grafana/grafana/discussions?discussions_q=sort%3Atop+created%3A%3E%3D2020-11-11

https://github.com/grafana/grafana/discussions/41104

**Comment by torkelo** [comment]
> nervous about this. 

me too, I agree there are some important features that would be good to have before making it impossible to select the old panel.

**Comment by ryantxu** [comment]
Can we add the "migrate" button soon -- but not actually deprecate until we have better solutions for the top issues with the new timeseries?

**Comment by torkelo** [comment]
@ryantxu removed the deprecation state change, so this now just adds the migration button

**Comment by ryantxu** [comment]
I changed the language so we don't say it is deprecated -- we should be extra careful about how we communicate that.  @pkolyvas 

image


I also added a special case to the search so that searching for "graph" will show timeseries ;)

image

**Review by ryantxu — APPROVED** [review]

File: public/app/plugins/panel/graph/tab_display.html [hunk 1]
```diff
@@ -1,4 +1,17 @@
...
+    <h5>Migration</h5>
+    <p>
+      Consider switching to the timeseries panel.  It is a more capable and performant version of this panel.
+    </p>
+    <p>
+      <button class="btn btn-primary" ng-click="ctrl.migrateToReact()">
+        Migrate
+      </button>
+    </p>
+    <p>Some features like colored time regions and negative transforms are not supported in the new panel.</p>
```
Comment (leeoniya) [inline public/app/plugins/panel/graph/tab_display.html:12]:
```suggestion
    <p>Some features like colored time regions and negative transforms are not supported in the new panel yet.</p>    
```

**Pull request merged**